src/main/java/org/jboss/tm/TransactionManagerLocator.java

```diff
@@ -21,166 +21,138 @@
+     */
+    public static TransactionManagerLocator getInstance() {
+        return instance;
+    }
+
+    /**
+     * Locate the transaction manager
+     *
+     * @return the transaction manager
+     */
+    public static TransactionManager locateTransactionManager() {
+        return getInstance().locate();
+    }
+
+    /**
+     * Locate the transaction manager
+     *
+     * @param throwExceptionIfUnableToLocate if true and transaction manager cannot be located, a
+     *     RuntimeException is thrown, otherwise null is returned.
+     * @return the transaction manager
+     */
+    public static TransactionManager locateTransactionManager(
+            boolean throwExceptionIfUnableToLocate) {
+        return getInstance().locate(throwExceptionIfUnableToLocate);
+    }
+
+    /**
+     * Get the transaction manager
+     *
+     * @return the transaction manager
+     */
+    public TransactionManager getTransactionManager() {
+        return locate();
+    }
+
+    /**
+     * Locate the transaction manager
+     *
+     * @return the transaction manager
+     */
+    public TransactionManager locate() {
+        return locate(true);
+    }
+
+    /**
+     * Locate the transaction manager
+     *
+     * @param throwExceptionIfUnableToLocate if true and transaction manager cannot be located, a
+     *     RuntimeException is thrown, otherwise null is returned.
+     * @return the transaction manager
+     */
+    public TransactionManager locate(boolean throwExceptionIfUnableToLocate) {
+        if (tm != null) return tm;
+
+        TransactionManager result = tryJNDI();
+        if (result == null) result = usePrivateAPI();
+        if (result == null && throwExceptionIfUnableToLocate)
+            throw new RuntimeException("Unable to locate the transaction manager");
+
+        return result;
+    }
+
+    /**
+     * Locate the transaction manager in the well known jndi binding for JBoss
+     *
+     * @return the tm from jndi
+     */
+    protected TransactionManager tryJNDI() {
+        try {
+            InitialContext ctx = new InitialContext();
+            tm = (TransactionManager) ctx.lookup(JNDI_NAME);
+            if (log.isTraceEnabled()) log.trace("Got a transaction manager from jndi " + tm);
+        } catch (NamingException e) {
+            log.debugf("Unable to lookup JNDI name %s reason %s", JNDI_NAME, e.getMessage());
+        }
+        return tm;
+    }
+
+    /**
+     * Use the private api
+     *
+     * <p>
+     *
+     * <p>This is a fallback method for non JBossAS use.
+     *
+     * @return the tm from the private api
+     */
+    protected TransactionManager usePrivateAPI() {
+        try {
+            Class<?> clazz = Class.forName("org.jboss.tm.TxManager");
+            Method method = clazz.getMethod("getInstance");
+            TransactionManager result = (TransactionManager) method.invoke(null, null);
+            log.warn("Using the old JBoss transaction manager " + tm);
```

Comment:
Remark: The original author did not much care for blank lines to help with readability, I am surprised that our rules do not impact on this coding style.